components/stego/SteganographyWorkbench.tsx

```
@@ -12,11 +14,43 @@ export default function SteganographyWorkbench() {
   const [chiResult, setChiResult] = useState<ChiSquareResult | null>(null);
-  const [selectedPlane, setSelectedPlane] = useState<number>(0);
+  const [selectedSample, setSelectedSample] = useState<'gradient' | 'geometric' | 'colorbars'>('gradient');
+  const [coverImageData, setCoverImageData] = useState<ImageData | null>(null);
+  const [stegoImageData, setStegoImageData] = useState<ImageData | null>(null);
   
   const canvasRef = useRef<HTMLCanvasElement | null>(null);
   const fileInputRef = useRef<HTMLInputElement | null>(null);
 
+  // Load initial sample on mount
+  useEffect(() => {
+    loadSampleCover(selectedSample);
```

ESLint

[error] Error: Cannot access variable before it is declared

`loadSampleCover` is accessed before it is declared, which prevents the earlier access from updating when this value changes over time.

/home/jailuser/git/components/stego/SteganographyWorkbench.tsx:26:5
  24 |   // Load initial sample on mount
  25 |   useEffect(() => {
> 26 |     loadSampleCover(selectedSample);
     |     ^^^^^^^^^^^^^^^ `loadSampleCover` accessed before it is declared
  27 |   }, []);
  28 |
  29 |   const loadSampleCover = (sampleType: 'gradient' | 'geometric' | 'colorbars') => {

/home/jailuser/git/components/stego/SteganographyWorkbench.tsx:29:3
  27 |   }, []);
  28 |
> 29 |   const loadSampleCover = (sampleType: 'gradient' | 'geometric' | 'colorbars') => {
     |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 30 |     try {
     | ^^^^^^^^^
> 31 |       const sampleData = generateSampleCover(sampleType, 400, 300);
     …
     | ^^^^^^^^^
> 51 |     }
     | ^^^^^^^^^
> 52 |   };
     | ^^^^^ `loadSampleCover` is declared here
  53 |
  54 |   const handleImageUpload = (e: React.ChangeEvent<HTMLInputElement>) => {
  55 |     const file = e.target.files?.[0];

(react-hooks/immutability)